Test failures with recent numpy versions

During a rebuild of the Debian archive we had 4 test failures.
The tests were previously working and we thing that the issue could related to update of numpy (currently 1.23.5) or some of the other dependencies.
Please find the relevant part of the log below.

I see that the `test_reverse_terra_with_zero_iterations` has been already fixed in git.
For other test I see that the error only exceeds slightly the tolerance so I think that it should be fine to relax a little bit.
Some guidance on how to solve/workaround the issue would be appreciated.
If considered useful, I could open a PR with the proposed fix.


```
> ============================= test session starts ==============================
> platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0+repack
> rootdir: /<<PKGBUILDDIR>>
> plugins: astropy-header-0.2.2, doctestplus-0.12.1, mock-3.8.2, filter-subpackage-0.1.1, cov-4.0.0, arraydiff-0.5.0, hypothesis-6.60.0, astropy-0.10.0, openfiles-0.5.0, remotedata-0.3.3
> collected 521 items / 226 deselected / 295 selected
> 
> skyfield/tests/test_against_novas.py ................................... [ 11%]
> ........................................                                 [ 25%]
> skyfield/tests/test_api.py .                                             [ 25%]
> skyfield/tests/test_broadcastability.py .                                [ 26%]
> skyfield/tests/test_constellations.py .                                  [ 26%]
> skyfield/tests/test_curves.py .                                          [ 26%]
> skyfield/tests/test_data_iers.py ..                                      [ 27%]
> skyfield/tests/test_earth_satellites.py ......                           [ 29%]
> skyfield/tests/test_earthlib.py F                                        [ 29%]
> skyfield/tests/test_elementslib.py ......F.F....                         [ 34%]
> skyfield/tests/test_frames.py ....                                       [ 35%]
> skyfield/tests/test_functions.py .....                                   [ 37%]
> skyfield/tests/test_geometry.py .                                        [ 37%]
> skyfield/tests/test_io.py .......                                        [ 40%]
> skyfield/tests/test_io_parsing.py ........                               [ 42%]
> skyfield/tests/test_keplerlib.py ...................                     [ 49%]
> skyfield/tests/test_positions.py .........F..                            [ 53%]
> skyfield/tests/test_satellite_events.py ..                               [ 53%]
> skyfield/tests/test_searchlib.py ......                                  [ 55%]
> skyfield/tests/test_strs_and_reprs.py ....                               [ 57%]
> skyfield/tests/test_text_pck.py ..                                       [ 57%]
> skyfield/tests/test_timelib.py ...........s...................s......... [ 71%]
> ................................                                         [ 82%]
> skyfield/tests/test_topos.py ...................................         [ 94%]
> skyfield/tests/test_trigonometry.py .                                    [ 94%]
> skyfield/tests/test_units.py ...............                             [100%]
> 
> =================================== FAILURES ===================================
> ___________________ test_reverse_terra_with_zero_iterations ____________________
> 
>     def test_reverse_terra_with_zero_iterations():
>         # With zero iterations, should return "geocentric" rather than
>         # "geodetic" (="correct") longitude and latitude.
>         lat, lon, elevation = reverse_terra(array([1, 0, 1]), 0, iterations=0)
>         assert abs(lat - tau / 8) < 1e-16
>         assert lon == 0.0
> >       assert abs(elevation - (AU_M * sqrt(2) - ERAD)) < 1e-16
> E       assert 6.103515625e-05 < 1e-16
> E        +  where 6.103515625e-05 = abs((211556959509.4766 - ((149597870700 * 1.4142135623730951) - 6378136.6)))
> E        +    where 1.4142135623730951 = sqrt(2)
> 
> skyfield/tests/test_earthlib.py:10: AssertionError
> ________________________________ test_parabolic ________________________________
> 
> ts = <skyfield.timelib.Timescale object at 0x7fbaecba04f0>
> 
>     def test_parabolic(ts):
>         e = 1
>         check_orbit(300000, e, angles1[:4], 0, 4, 3, ts)
>         check_orbit(300000, e, 2, angles1, 4, 3, ts)
> >       check_orbit(300000, e, 2, 3, angles1, 3, ts)
> 
> skyfield/tests/test_elementslib.py:512: 
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> skyfield/tests/test_elementslib.py:428: in check_orbit
>     compare(elements.semi_latus_rectum.km, p, 1e-9)
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> 
> value = array([300000., 300000., 300000., 300000., 300000., 300000., 300000.,
>        300000.])
> expected_value = 300000, epsilon = 1e-09, mod = False
> 
>     def compare(value, expected_value, epsilon, mod=False):
>         """Compares value to expected value, and works if one or both are arrays.
>     
>         Also allows epsilon to be an array.
>     
>         If mod==True, then compare(0, tau, 0) is True.
>         """
>         if mod:
>             diff = normpi(value - expected_value)
>         else:
>             diff = value - expected_value
>     
>         if hasattr(value, '__len__') or hasattr(expected_value, '__len__'):
>             if hasattr(epsilon, '__len__'):
>                 assert (abs(diff) <= epsilon).all()
>             else:
> >               assert max(abs(diff)) <= epsilon
> E               assert 1.0477378964424133e-09 <= 1e-09
> E                +  where 1.0477378964424133e-09 = max(array([0.00000000e+00, 4.65661287e-10, 0.00000000e+00, 1.04773790e-09,\n       0.00000000e+00, 1.16415322e-10, 1.74622983e-10, 2.91038305e-10]))
> E                +    where array([0.00000000e+00, 4.65661287e-10, 0.00000000e+00, 1.04773790e-09,\n       0.00000000e+00, 1.16415322e-10, 1.74622983e-10, 2.91038305e-10]) = abs(array([ 0.00000000e+00, -4.65661287e-10,  0.00000000e+00, -1.04773790e-09,\n        0.00000000e+00, -1.16415322e-10,  1.74622983e-10,  2.91038305e-10]))
> 
> skyfield/tests/test_elementslib.py:51: AssertionError
> _____________________________ test_parabolic_polar _____________________________
> 
> ts = <skyfield.timelib.Timescale object at 0x7fbaecdfdde0>
> 
>     def test_parabolic_polar(ts):
>         e = 1
>         i = pi/2
>         check_orbit(300000, e, i, angles1, 2, 3, ts)
> >       check_orbit(300000, e, i, 1, angles1, 3, ts)
> 
> skyfield/tests/test_elementslib.py:539: 
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> skyfield/tests/test_elementslib.py:428: in check_orbit
>     compare(elements.semi_latus_rectum.km, p, 1e-9)
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> 
> value = array([300000., 300000., 300000., 300000., 300000., 300000., 300000.,
>        300000.])
> expected_value = 300000, epsilon = 1e-09, mod = False
> 
>     def compare(value, expected_value, epsilon, mod=False):
>         """Compares value to expected value, and works if one or both are arrays.
>     
>         Also allows epsilon to be an array.
>     
>         If mod==True, then compare(0, tau, 0) is True.
>         """
>         if mod:
>             diff = normpi(value - expected_value)
>         else:
>             diff = value - expected_value
>     
>         if hasattr(value, '__len__') or hasattr(expected_value, '__len__'):
>             if hasattr(epsilon, '__len__'):
>                 assert (abs(diff) <= epsilon).all()
>             else:
> >               assert max(abs(diff)) <= epsilon
> E               assert 1.57160684466362e-09 <= 1e-09
> E                +  where 1.57160684466362e-09 = max(array([1.16415322e-10, 0.00000000e+00, 1.74622983e-10, 1.57160684e-09,\n       0.00000000e+00, 0.00000000e+00, 1.74622983e-10, 1.16415322e-10]))
> E                +    where array([1.16415322e-10, 0.00000000e+00, 1.74622983e-10, 1.57160684e-09,\n       0.00000000e+00, 0.00000000e+00, 1.74622983e-10, 1.16415322e-10]) = abs(array([-1.16415322e-10,  0.00000000e+00,  1.74622983e-10,  1.57160684e-09,\n        0.00000000e+00,  0.00000000e+00,  1.74622983e-10, -1.16415322e-10]))
> 
> skyfield/tests/test_elementslib.py:51: AssertionError
> ________________________ test_velocity_in_ITRF_to_GCRS2 ________________________
> 
>     def test_velocity_in_ITRF_to_GCRS2():
>         # TODO: Get test working with these vectors too, showing it works
>         # with a non-zero velocity vector, but in that case the test will
>         # have to be fancier in how it corrects.
>         # r = np.array([(1, 0, 0), (1, 1 / DAY_S, 0)]).T
>         # v = np.array([(0, 1, 0), (0, 1, 0)]).T
>     
>         ts = api.load.timescale()
>         t = ts.utc(2020, 7, 17, 8, 51, [0, 1])
>         r = np.array([(1, 0, 0), (1, 0, 0)]).T
>         v = np.array([(0, 0, 0), (0, 0, 0)]).T
>     
>         r, v = ITRF_to_GCRS2(t, r, v, True)
>     
>         # Rotate back to equinox-of-date before applying correction.
>         r = mxv(t.M, r)
>         v = mxv(t.M, v)
>     
>         r0, r1 = r.T
>         v0 = v[:,0]
>     
>         # Apply a correction: the instantaneous velocity does not in fact
>         # carry the position in a straight line, but in an arc around the
>         # origin; so use trigonometry to move the destination point to where
>         # linear motion would have carried it.
>         angvel = (t.gast[1] - t.gast[0]) / 24.0 * tau
>         r1 = mxv(rot_z(np.arctan(angvel) - angvel), r1)
>         r1 *= np.sqrt(1 + angvel*angvel)
>     
>         actual_motion = r1 - r0
>         predicted_motion = v0 / DAY_S
>     
>         relative_error = (length_of(actual_motion - predicted_motion)
>                           / length_of(actual_motion))
>     
>         acceptable_error = 4e-12
> >       assert relative_error < acceptable_error
> E       assert 9.700550724172958e-12 < 4e-12
> 
> skyfield/tests/test_positions.py:188: AssertionError
> =========================== short test summary info ============================
> FAILED skyfield/tests/test_earthlib.py::test_reverse_terra_with_zero_iterations
> FAILED skyfield/tests/test_elementslib.py::test_parabolic - assert 1.04773789...
> FAILED skyfield/tests/test_elementslib.py::test_parabolic_polar - assert 1.57...
> FAILED skyfield/tests/test_positions.py::test_velocity_in_ITRF_to_GCRS2 - ass...
> =========== 4 failed, 289 passed, 2 skipped, 226 deselected in 5.08s ===========
```

--- a/skyfield/tests/test_elementslib.py
+++ b/skyfield/tests/test_elementslib.py
@@ -423,7 +423,7 @@ def check_orbit(p, e, i, Om, w, v, ts):
                                   time, mu)
     check_types(elements, length)
     compare(time, elements.time, 1e-9)
-    compare(elements.semi_latus_rectum.km, p, 1e-9)
+    compare(elements.semi_latus_rectum.km, p, 1e-8)
     compare(elements.eccentricity, e, 1e-14)
     compare(elements.inclination.radians, i, 1e-14, mod=True)
     compare(elements.longitude_of_ascending_node.radians, Om, 1e-14, mod=True)

--- a/skyfield/tests/test_planetarylib.py
+++ b/skyfield/tests/test_planetarylib.py
@@ -45,14 +45,14 @@ def test_frame_rotation_matrices():
     desired_rate = np.array(desired_rate) * DAY_S
 
     R = frame.rotation_at(ts.tdb_jd(tdb))
-    assert (R == desired_rotation).all()  # Boom.
+    assert abs(R - desired_rotation).max() < 3e-16
 
     R2, Rv = frame.rotation_and_rate_at(ts.tdb_jd(tdb))
     assert (R == R2).all()
     if IS_32_BIT:
         assert abs(Rv - desired_rate).max() < 3e-26
     else:
-        assert (Rv == desired_rate).all()  # Boom.
+        assert abs(Rv - desired_rate).max() < 3e-17
 
     # Second, a moment when the angle W is more than 2500 radians.
 
@@ -107,14 +107,14 @@ def test_frame_rotation_matrices():
     if IS_32_BIT:
         assert abs(R - desired_rotation).max() < 2e-16
     else:
-        assert (R == desired_rotation).all()
+        assert abs(R - desired_rotation).max() < 4e-16
 
     R2, Rv = frame.rotation_and_rate_at(ts.tdb_jd(tdb))
     assert (R == R2).all()
     if IS_32_BIT:
         assert abs(Rv - desired_rate).max() < 2e-18
     else:
-        assert (Rv == desired_rate).all()
+        assert abs(Rv - desired_rate).max() < 6e-17
 
 def test_rotating_vector_into_frame():
     et_seconds = 259056665.1855896

--- a/skyfield/tests/test_positions.py
+++ b/skyfield/tests/test_positions.py
@@ -139,7 +139,7 @@ def test_position_of_radec():
     ra, dec, distance = p.radec(epoch=epoch)
     assert abs(ra.hours) < 1e-12
     assert abs(dec.degrees) < 1e-12
-    assert abs(distance.au - 1) < 1e-16
+    assert abs(distance.au - 1) < 3e-16
 
 def test_position_from_radec():
     # Only a couple of minimal tests, since the routine is deprecated.

--- a/skyfield/tests/test_positions.py
+++ b/skyfield/tests/test_positions.py
@@ -184,7 +184,7 @@ def test_velocity_in_ITRF_to_GCRS2():
     relative_error = (length_of(actual_motion - predicted_motion)
                       / length_of(actual_motion))
 
-    acceptable_error = 4e-12
+    acceptable_error = 1e-11
     assert relative_error < acceptable_error
 
 def test_light_time_method():

